samples/java/src/main/java/com/modulr/api/ModulrApiAuth.java

```diff
@@ -0,0 +1,133 @@
+package com.modulr.api;
+
+import javax.crypto.Mac;
+import javax.crypto.spec.SecretKeySpec;
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+import java.security.SignatureException;
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.*;
+
+public class ModulrApiAuth {
+    private static final String HMAC_SHA1_ALGORITHM = "HmacSHA1";
+    private static final String DATE_PATTERN = "EEE, dd MMM yyyy HH:mm:ss z";
+    private final String secret;
+    private final String token;
+    private String nonce;
+    private Date date;
+    private Boolean retry = false;
+
+    private String lastGeneratedHmac;
+
+    public ModulrApiAuth(String token, String secret, String nonce) {
+        this.token = token.trim();
+        this.secret = secret.trim();
+        this.nonce = nonce.trim();
+        this.date = new Date();
+    }
+
+    public Map<String, String> getApiAuthHeaders() {
+        final Map<String, String> headerParams = new HashMap<>();
+        try {
+            String hmac = generateHmac();
+
+            headerParams.put("Authorization", formatAuthHeader(this.token, hmac));
+            headerParams.put("Date", getFormattedDate(this.date));
+            headerParams.put("x-mod-nonce", this.nonce);
+            headerParams.put("x-mod-retry", String.valueOf(this.retry));
+        } catch (SignatureException e) {
+            e.printStackTrace();
+        }
+
+        return headerParams;
+    }
+
+    public String generateHmac() throws SignatureException {
+        final String hmac;
+        if (this.retry) {
```

Comment:
This doesnt seem correct. If retry is true then the nonce should be same as the last time, but the time would have moved on so need to generate new hmac.  Without this, retry can only be used within clock skew tolerance which I think is set to 15 minutes only,.